### PR TITLE
Persistent params in links support

### PR DESCRIPTION
**Pull request by MartinMystikJonas**
Partialy solve #198

### CODE REVIEW

File: src/LinkProcessor/LinkParamsProcessor.php [hunk 1]
```diff
@@ -83,6 +86,16 @@ public function process(string $class, string $method, array $params): array
             }
         }
 
+        // remove persistent parameters
```
Comment (lulco) [inline src/LinkProcessor/LinkParamsProcessor.php:89]: We should remove them only if they are not part of method params, I don't see the check here.

Let's say we have persistent param $p, and some actionDefault($a, $b, $p) <- here we should keep $p in url params

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte [hunk 2]
```diff
@@ -85,3 +85,7 @@
 <a href="{link this#anchor}">Link to this with anchor</a>
 <a href="{link this, foo: bar, bar: baz}">Link to this with params</a>
 <a href="{link this#anchor, foo: bar, bar: baz}">Link to this with params and anchor</a>
+
+<a href="{link create persistent1: "persistent"}">With persistent annotation</a>
```
Comment (lulco) [inline tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte:89]: We need also create tests where persisten parameter is part of action e.g. `actionSomething($persistent)` <- here we want to keep persistent parameter.

File: src/LinkProcessor/LinkParamsProcessor.php (hunk 1 above)
Comment (MartinMystikJonas) [reply src/LinkProcessor/LinkParamsProcessor.php:89]: I think it is processed in previous block if they are part of method params and mobed to $methodParams

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte (hunk 2 above)
Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte:89]: Ok. Have to check how this conflict is handled by nette

**Comment by MartinMystikJonas** [comment]
Idea how to best have dirrerent expexted errors based on Php version?

File: src/LinkProcessor/LinkParamsProcessor.php (hunk 1 above)
Comment (lulco) [reply src/LinkProcessor/LinkParamsProcessor.php:89]: Yes, but here you will remove them

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte (hunk 2 above)
Comment (lulco) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte:89]: Do you think nette get persistent parameters in link as some special parameter?

File: src/LinkProcessor/LinkParamsProcessor.php (hunk 1 above)
Comment (MartinMystikJonas) [reply src/LinkProcessor/LinkParamsProcessor.php:89]: From transferedparams not from methodparams

Comment (lulco) [reply src/LinkProcessor/LinkParamsProcessor.php:89]: yes, and then they are not returned to LinkProcessor

**Comment by lulco** [comment]
> Idea how to best have dirrerent expexted errors based on Php version?

I think I need more context

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte (hunk 2 above)
Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/templates/Links/default.latte:89]: No. But i am not sure how request is provesed when method param and presistent param conflicts.

**Comment by MartinMystikJonas** [comment]
> > Idea how to best have dirrerent expexted errors based on Php version?
> 
> I think I need more context

In PHP7.4 attributes do not exists so persistent paramater with attribute is seen as not-persistent. There is one more expected error. I fixed it by creating expected errors before abalyse call and conditionally add item there.

**Comment by lulco** [comment]
If user analyses his php8.1 code with 7.4, he will get error 🤷 I would ignore this case

**Comment by MartinMystikJonas** [comment]
Well I do not think it is something we should solve. If PHPStan runs on older version that app is imho something that should be fixed by user.

**Review by lulco — APPROVED** [review]

**Pull request merged**